utils/management_commands.py

```diff
@@ -0,0 +1,59 @@
+#
+# Freesound is (c) MUSIC TECHNOLOGY GROUP, UNIVERSITAT POMPEU FABRA
+#
+# Freesound is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Freesound is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     See AUTHORS file.
+#
+
+import json
+import logging
+import os
+import sys
+import time
+
+from django.core.management.base import BaseCommand
+
+
+commands_logger = logging.getLogger('commands')
+
+
+class LoggingBaseCommand(BaseCommand):
+
+    start_time = None
+    command_name = None
+
+    def find_command_name(self):
+        for count, arg in enumerate(sys.argv):
+            if arg == 'manage.py':
+                return sys.argv[count + 1]
+        return None
+
+    def log_start(self, data=None):
+        self.start_time = time.time()
+        self.command_name = self.find_command_name()
+        if data is None:
+            data = {}
+        data['command'] = self.command_name
+        commands_logger.info('Started management command ({0})'.format(json.dumps(data)))
+
+    def log_end(self, data=None):
+        command_name = os.path.basename(__file__)
+        if data is None:
+            data = {}
+        if self.start_time:
+            data['work_time'] = time.time() - self.start_time
+        data['command'] = self.command_name or self.find_command_name()
+        commands_logger.info('Finished management command ({0})'.format(json.dumps(data)))
```

Comment:
I'm not a huge fan of this format of message, because if there is a lot of data here, it's difficult to read. Is this specifically for graylog? If we're going to use a helper tool like this maybe it's better to have it output something easily readable by people, and send a message directly to graylog using a dedicated logger in a format that it understands.

Reply:
This is for graylog yes. We use it in many places. To be honest I think this data is not to be read a lot by humans in log files. But if it needs to be read it's not too bad because most of the times its only a property or two that we set here. The idea however is that these things will be inspected in graylog, and there we can format them nicely. Using a different format for printing on screen means more lines of code which I don't think will be really needed. For the commands we typically want to see the output on the console, we also have a console_logger which prints the info (eg the report_index_statuses command)